internal/pkg/cri/pods.go

```diff
@@ -79,3 +81,92 @@ func (c *Client) PodSandboxStatus(ctx context.Context, podSandBoxID string) (*ru
 
 	return resp.Status, resp.Info, nil
 }
+
+// RemovePodSandboxesWithNetworkMode removes all pods with the specified network mode.
+func (c *Client) RemovePodSandboxesWithNetworkMode(mode runtimeapi.NamespaceMode) (err error) {
+	ctx := context.Background()
+
+	pods, err := c.ListPodSandbox(ctx, nil)
+	if err != nil {
+		return err
+	}
+
+	var g errgroup.Group
+
+	for _, pod := range pods {
+		pod := pod // https://golang.org/doc/faq#closures_and_goroutines
+
+		status, _, err := c.PodSandboxStatus(ctx, pod.GetId())
+		if err != nil {
+			return err
+		}
+
+		networkMode := status.GetLinux().GetNamespaces().GetOptions().GetNetwork()
+
+		if networkMode != mode {
+			continue
+		}
+
+		g.Go(func() error {
+			if err := remove(ctx, c, pod); err != nil {
+				return err
```

Comment:
looks like could be `return remove(ctx, c, pod)`

plus it needs to wrap variables, as `pod` is `for` loop variable

Reply:
A bit further up we use `pod := pod // https://golang.org/doc/faq#closures_and_goroutines`. A little trick I picked up not too long ago.

Reply:
oh, missed that, sorry for the noise!

Reply:
Moved to using `return remove(ctx, c, pod)`.